src/java/com/zimbra/qa/selenium/framework/core/ExecuteHarnessMain.java

```diff
@@ -1327,6 +1329,117 @@ private boolean parseArgs(String arguments[]) throws ParseException, HarnessExce
+
+		// Create harness log folder and file
+		fHarnessLogFileFolder = new File(sHarnessLogFileFolderPath);
+		if (!fHarnessLogFileFolder.exists()) {
+			fHarnessLogFileFolder.mkdirs();
+		}
+
+		try {
+			if (fHarnessLogFile.createNewFile()) {
+				logger.info("Creating harness log file: " + sHarnessLogFileName);
+			}
+		} catch (IOException e1) {
+			e1.printStackTrace();
+		}
+
+		logInfo = "-------------- Zimbra pre-configuration and required setup for " + project + " project --------------";
+		logger.info(logInfo);
+		Files.write(pHarnessLogFilePath, Arrays.asList(logInfo), Charset.forName("UTF-8"), StandardOpenOption.APPEND);
+
+		// Get all zimbra servers
+		logger.info("Get all zimbra servers");
+		for (String noOfZimbraServers : CommandLineUtility.runCommandOnZimbraServer("zmprov -l gas | wc -l")) {
+			totalZimbraServers = Integer.parseInt(noOfZimbraServers);
+		}
+
+		// Get all store servers
+		logger.info("Get all store servers");
+		storeServers = CommandLineUtility.runCommandOnZimbraServer("zmprov -l gas mailbox");
+
+		if (totalZimbraServers == 0 || storeServers.isEmpty()) {
+			seleniumService.stopSeleniumExecution();
+			logInfo = "Couldn't get total zimbra servers (" + totalZimbraServers + ") or store servers (" + storeServers + ") using CLI commands";
+			logger.info(logInfo);
+			Files.write(pHarnessLogFilePath, Arrays.asList(logInfo), Charset.forName("UTF-8"), StandardOpenOption.APPEND);
+			System.exit(0);
+		} else {
+			logInfo = "No. of zimbra server(s): " + totalZimbraServers;
+			logger.info(logInfo);
+			Files.write(pHarnessLogFilePath, Arrays.asList(logInfo), Charset.forName("UTF-8"), StandardOpenOption.APPEND);
+
+			logInfo = "Store server(s): " + storeServers;
+			logger.info(logInfo);
+			Files.write(pHarnessLogFilePath, Arrays.asList(logInfo), Charset.forName("UTF-8"), StandardOpenOption.APPEND);
+		}
+
+		// Grant createDistList right to domain
+		logInfo = "Grant createDistList right to domain using STAF";
+		logger.info(logInfo);
+		Files.write(pHarnessLogFilePath, Arrays.asList(logInfo), Charset.forName("UTF-8"), StandardOpenOption.APPEND);
+		CommandLineUtility.runCommandOnZimbraServer("zmprov grr domain " + ConfigProperties.getStringProperty("testdomain")
+						+ " dom " + ConfigProperties.getStringProperty("testdomain") + " createDistList");
+
+		// Disable zimbraSmimeOCSPEnabled attribute for S/MIME
+		logInfo = "Disable zimbraSmimeOCSPEnabled attribute for S/MIME using STAF";
+		logger.info(logInfo);
+		Files.write(pHarnessLogFilePath, Arrays.asList(logInfo), Charset.forName("UTF-8"), StandardOpenOption.APPEND);
+		CommandLineUtility.runCommandOnZimbraServer("zmprov mcf zimbraSmimeOCSPEnabled FALSE");
+
+		// Universal UI project settings
+		if (project.contains("universal")) {
+			String universalUITheme = "clarity";
+
+			Boolean themeFound = false;
+			for (String serverUniversalUITheme : CommandLineUtility.runCommandOnZimbraServer("ls /opt/zimbra/jetty/webapps/zimbra/skins")) {
```

Comment:
ls /opt/zimbra/jetty/webapps/zimbra/skins/ | grep clarity
would be easier operation wise

Reply:
Agreed, done.